src/Microsoft.DotNet.XHarness.iOS.Shared/Listeners/SimpleTcpListener.cs

```diff
@@ -70,6 +95,75 @@ protected override void Start()
             }
         }
 
+        void StartTcpTunnel()
+        {
+            if (!TunnelHoleThrough.Task.Result)
+            { // do nothing until the tunnel is ready
+                throw new InvalidOperationException("Tcp tunnel could not be initialized.");
+            }
+            bool processed;
+            try
+            {
+                int timeout = 100;
+                var watch = new System.Diagnostics.Stopwatch();
+                watch.Start();
+                while (true)
+                {
+                    try
+                    {
+                        client = new TcpClient("localhost", Port);
+                        Log.WriteLine("Test log server listening on: {0}:{1}", Address, Port);
+                        // let the device know we are ready!
+                        var stream = client.GetStream();
+                        var ping = Encoding.UTF8.GetBytes("ping");
+                        stream.Write(ping, 0, ping.Length);
+                        break;
+
+                    }
+                    catch (SocketException ex)
+                    {
+                        if (timeout == 100 && watch.ElapsedMilliseconds > 20000)
+                        {
+                            timeout = 250; // Switch to a 250ms timeout after 20 seconds
+                        }
+                        else if (timeout == 250 && watch.ElapsedMilliseconds > 120000)
+                        {
+                            // Give up after 2 minutes.
+                            throw ex;
+                        }
+                        Thread.Sleep(timeout);
```

Comment:
Maybe Log a retry and after how long? Would be nice to know from logs how long these things take to connect